src/utils/onCookieChange.ts

```diff
@@ -0,0 +1,28 @@
+/**
+ * An event listener for cookie changes.
+ * @param {Function} callback A callback function to be called when the cookie changes
+ * @param {number} interval number of milliseconds to wait between each check
+ */
+
+export function onCookieChange(
+    callback: (event: {
+        previousCookies: string;
+        currentCookies: string;
+    }) => void,
+    interval = 1000
+) {
+    let lastCookie = document.cookie;
+    setInterval(() => {
```

Comment:
Are we clearing this interval somewhere? or will it cause multiple intervals set on window?

Reply:
It is a dead code. I'll remove this code with the next push.